### ISSUE TITLE
Browser dock size resets on fresh install

### ISSUE THREAD

**Issue by flaeri** [obsproject/obs-studio#5560]
### Operating System Info

Windows 10

### OBS Studio Version

27.1.3

### OBS Studio Log URL

https://obsproject.com/logs/dtWIp5yczinM06Fp

### Expected Behavior

On a fresh install, I would expect that when adding custom browser docks, and resize them to a comfortable size, that the docks would stay that size, and not reset back to the tiny size you get when you first dock them.

### Current Behavior

Upon adding 2 custom browser docks, and attaching them to the right and left size, then resizing them to be larger, they will reset back to their original tiny size upon restarting OBS.

### Steps to Reproduce

1. Fresh install of OBS (or clean out config/global.ini)
2. Add 2 fresh custom browser sources
a. for instance <https://google.com> and <https://obsproject.com>
3. "Attach" / dock one on each side (right / left). They will be very slim
4. Resize both to be larger
5. Restart OBS
6. Observe how the docks are reset back to their original size (very slim)

### Anything else we should know?

I believe that this behavior is caused by the "source toolbar".
If you repeat the same repro step, but before step 2, you first disable the source toolbar, the behavior will not happen.

I also *feel* like there is something related to the size of OBS/maximize state, possibly in conjunction with the source toolbar. Resizing the window, maximizing, resizing, and maximizing sometimes corrects the behavior of the browser docks resetting.

Thanks to @lindenkron who brought up the bug, and gave me repro step, and additional information.

**Comment by Azeirah** [comment]
![obs docked sites](https://user-images.githubusercontent.com/4040870/143284440-02b2548f-f0e6-4cd4-8c42-e6545f85587a.png)

I'm trying to replicate the issue on OBS 27.1.3, but I'm not seeing the behaviour you're describing. By 'docking', do you mean the two webpages visible in the scene? I attached obs to the left and chrome to the right, both of them stayed the same size.

I tried this on both windows and macOS.

IE, what I did was

1. Reset OBS config
2. Under the 'sources' press + and select 'browser'
3. Add one for obsproject.com
4. Repeat 2,3 for google.com
5. Drag the google.com window to the right until it snaps
6. Drag the obsproject.com window to the left until it snaps

No resizing happening whatsoever on my end. Am I misunderstanding you?

**Comment by flaeri** [comment]
Yeah, thats my bad. Writing "Browser Sources", when I meant "browser docks"
![image](https://user-images.githubusercontent.com/50419942/143300709-47312cd8-25a7-432e-8bf3-0a1dbe2b5536.png)

![image](https://user-images.githubusercontent.com/50419942/143300794-b46021fb-ab4b-4564-bba1-04748843a305.png)

**Comment by Azeirah** [comment]
Ahh thank you, I can reproduce the issue now. It's definitely related to the source toolbar.

When you make the docks larger and _then_ turn **on** the source toolbar, the two docked windows are squished back to their original size.

**Comment by Azeirah** [comment]
It looks like #5412 fixes this issue, I tested it with that patch applied and the resizing bug doesn't happen anymore.

I think that means this can be closed?

**Comment by flaeri** [comment]
hmm, yeah, it certainly helps, that for sure. I guess it depends on how you look at it, and what is the expected behavior. Even with the new build from #5412, it will still resize, just not in as a drastic of a way.

With the new build, if I do the same steps, and set both sides to be about 400px (and keep the source toolbar on), and then relaunch, the will resize down to about 300px (probably still related to the size of the source toolbar).

If I disabled the toolbar, they will not change from their initial 400px width. I would consider this a large improvement, and the effect might now be considered negligible. I think I would still say that this is probably unwanted/unexpected behavior.

Should the dev team decide that this is good enough, feel free to close, but the initial report was mainly upset that the browser docks were not the same size they left them at (which still holds true).

**Comment by Azeirah** [comment]
As far as I can see, there are now two issues caused by docked browser windows interacting with the source toolbar.

The first is the original issue, but less bad (like you described).

The second is that when additional icons are added to the source toolbar while it is open, the docked browser window will be forcefully resized to accomodate for the new icon.

I think this is a separate issue, since one part of the original issue actually _has_ been resolved.

Namely:

If you have the source toolbar disabled, and the two docked browser windows at maximum size, and then proceed to open the source toolbar, it used to resize the browser windows before the merge.

Now, it stays the exact same size before it was closed. This is the behavior you'd expect when OBS starts up too, but for an unknown reason this is not what happens.

My idea is:

Split this up into two separate issues, and notify @gxalpha that this merge has a total of two negative and one positive side effect on dock resizing behavior.

Issue 1: When OBS is restarted, the source toolbar resizes docked browser windows

Issue 2: When two docked browser windows are at maximum size, adding a button to the source toolbar resizes a docked browser window.

**Comment by lindenkron** [comment]
I was the person who initially had issues with the resizing.

While I cannot speak for what the person maintaining the Source Toolbar initially wanted to do with it, I can say that from a user perspective;
**Anything** that without influence changes or manipulates specific user setups is something that should be avoided at all cost. It leads to nothing but nuisances.

You could argue that it's a minor thing in the scope of things; and that majority of users may never encounter this issue.

It still stands that any UX that allows customization should not inadvertently change those customizations without the users consent, since it negates the point of having it be customizable in the first place.

**Comment by Dilzze** [comment]
Sorry for my english its the same problem ? https://obsproject.com/forum/threads/expand-my-chat-box-mixer-sources-screen-windows-panels-reseting-in-diferen-position.156751/

**Comment by flaeri** [comment]
After the source toolbar was changed so that the it was collapsible when the window is small enough, I personally feel like the original issue was fixed. Closing as per https://github.com/obsproject/obs-studio/issues/5560#issuecomment-979431649
